### PR TITLE
feat: adding authorization header to web-post

### PR DESCRIPTION
**Pull request by doriaviram**
Adding the ability to add Authorization header to the POST request on web-post module

### CODE REVIEW

**Comment by doriaviram** [comment]
Hey @daniel-hauser, I don't know if you want to include this change, I needed to add it to ensure no one is pushing trash to my server.

BTW, this open source is amazing, great job

File: src/bot/storage/web-post.ts [hunk 1]
```diff
@@ -25,12 +26,18 @@ export class WebPostStorage implements TransactionStorage {
...
+      headers["Authorization"] = this.authorizationToken;
+    }
+
     const [response] = await Promise.all([
       fetch(this.url, {
         method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
+        headers: headers,
```
Comment (daniel-hauser) [inline src/bot/storage/web-post.ts]:
```suggestion
        headers: {
          "Content-Type": "application/json",
          ...(this.authorizationToken && {
            Authorization: this.authorizationToken,
          }),
        },
```

File: README.md [hunk 2]
```diff
@@ -194,9 +194,10 @@ The transactions will be sent as a JSON array in the body of the request with th
 
 Use the following env vars to setup:
 
-| env variable name | description        |
-| ----------------- | ------------------ |
-| `WEB_POST_URL `   | The URL to post to |
+| env variable name               | description                                          |
+|---------------------------------|------------------------------------------------------|
+| `WEB_POST_URL`                  | The URL to post to                                   |
+| `WEB_POST_AUTHORIZATION_TOKEN`  | Authorization header value like this: `Bearer *****` |
```
Comment (daniel-hauser) [inline README.md]:
```suggestion
| `WEB_POST_AUTHORIZATION_TOKEN`  | The Authorization header value (i.e. `Bearer *****`, but can use any schema) |
```

File: src/bot/storage/web-post.ts [hunk 3]
```diff
@@ -25,12 +26,18 @@ export class WebPostStorage implements TransactionStorage {
...
     logger(`Posting ${nonPendingTxns.length} transactions to ${this.url}`);
 
+    const headers = {
+      "Content-Type": "application/json",
+    };
+
+    if(!!this.authorizationToken) {
+      headers["Authorization"] = this.authorizationToken;
+    }
+
```
Comment (daniel-hauser) [inline src/bot/storage/web-post.ts]: If you inline:

```suggestion

```

**Comment by daniel-hauser** [comment]
@doriaviram Thanks for your PR!
I added some suggestions, will merge once committed/resolved

**Comment by doriaviram** [comment]
@daniel-hauser Fixed all comments, thanks

**Review by daniel-hauser — APPROVED** [review]

**Pull request merged**